__tests__/app.test.js

```diff
@@ -122,4 +122,44 @@ describe('/api/article',()=>{
             })
         })
     })
+})
+describe('/api/articles/:article_id/comments',()=>{
+    test('GET:200 should return an array of comments linked',()=>{
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body:{comments}})=>{
+            expect(comments).toHaveLength(11)
+            comments.forEach((comment)=>{
+                expect(comment).toEqual(
+                    expect.objectContaining({
+                        comment_id:expect.any(Number),
+                        votes:expect.any(Number),
+                        created_at:expect.any(String),
+                        author:expect.any(String),
+                        body:expect.any(String),
+                        article_id:expect.any(Number)
+                    })
+                )
+            })
+            expect(comments).toBeSortedBy('created_at',{
+                descending:true,})
+        })
+    })
+    test('GET: 404 when passed article_id is of a valid type but no comments are found',()=>{
+        return request(app)
+        .get('/api/articles/9999/comments')
+        .expect(404)
+        .then(({body})=>{
+            expect(body.msg).toBe('no comments found')
+        })
+    })
+    test('GET: 400 when passed article_id of invalid type object returned with an err message of Bad request',()=>{
+        return request(app)
+        .get('/api/articles/invalid_id/comments')
+        .expect(400)
+        .then(({body})=>{
+            expect(body.msg).toBe('Bad request')
+        })
```

Comment:
Great testing here so far 👏 the only other thing I think we could consider is what would happen if we queried an article that had no comments on it, we'd get an empty array back but this isn't an error, there's just no comments. So we would need to differentiate between this valid response and an error - one to come back to another time 👍